src/main/java/org/eln2/mc/common/CommonEvents.kt

```diff
@@ -38,4 +52,28 @@ object CommonEvents {
             }
         }
     }
+
+    @SubscribeEvent
+    fun onEntityJoinedWorld(event: EntityJoinWorldEvent) {
+        //Warn new users that we collect analytics
+        if (event.world.isClientSide && event.entity is Player) {
+            if (Eln2.config.enableAnalytics)  {
+                val acknowledgementFile = FMLPaths.CONFIGDIR.get().resolve("ElectricalAge/acknowledgements.yml").toFile()
+                println("Checking for file $acknowledgementFile")
+                if (!acknowledgementFile.exists()) {
+                    acknowledgementFile.parentFile.mkdirs()
+                    acknowledgementFile.createNewFile()
+                    acknowledgementFile.writeText(Yaml.default.encodeToString(AnalyticsAcknowledgementsData.serializer(), AnalyticsAcknowledgementsData(mutableMapOf())))
+                }
+                val uuid: String = event.entity.uuid.toString()
+                val acknowledgements: AnalyticsAcknowledgementsData = Yaml.default.decodeFromStream(AnalyticsAcknowledgementsData.serializer(), acknowledgementFile.inputStream())
+                if (acknowledgements.entries.none { (k, _) -> k == uuid } || acknowledgements.entries[uuid]!! < (System.currentTimeMillis() / 1000L) - THIRTY_DAYS) {
+                    event.entity.sendMessage(TranslatableComponent("misc.eln2.acknowledge_analytics").withStyle(ChatFormatting.RED), Util.NIL_UUID)
+                    acknowledgements.entries[uuid] = System.currentTimeMillis() / 1000L
+                }
+
+                acknowledgementFile.writeText(Yaml.default.encodeToString(AnalyticsAcknowledgementsData.serializer(), acknowledgements))
+            }
+        }
```

Comment:
Need some checking for if the file is not writable or if the file read back is corrupted somehow (binary data, user messing with it, etc)